### PR TITLE
Working forms, buttons linking posts and comments and nesting structure

### PR DESCRIPTION
**Pull request by jamescavallo**
Root page now shows all the available courses, clicking on a course title brings you to the posts page for that course where you can either create a new post or click on an existing one. Creating a new post correctly puts that post under the current course. Clicking on a post brings you to that post's comments where you can also create a new comment.

### CODE REVIEW

File: app/helpers/comments_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (jpaints3) [inline app/helpers/comments_helper.rb:1]: Is this to help with the routing for displaying comments? Just trying to make sure I keep up with the code.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Comment < ApplicationRecord
-  validates :text_body
+  validate :text_body
```
Comment (jpaints3) [inline app/models/comment.rb:2]: What does this do? Are we using custom validations?

**Review by jpaints3 — APPROVED** [review]
Looks good to me. I have a couple of quick questions to increase my understanding of the code.

**Review by itsjoyce — APPROVED** [review]

File: app/helpers/comments_helper.rb (hunk 1 above)
Comment (jamescavallo) [reply app/helpers/comments_helper.rb:1]: No, this is autogenerated with rails g controller, not sure if we really need it

File: app/models/comment.rb (hunk 2 above)
Comment (jamescavallo) [reply app/models/comment.rb:2]: I kept getting this: `You need to supply at least one validation` I think it needs at least one validator following the line

Comment (jpaints3) [reply app/models/comment.rb]: Ok sounds good, yeah, I can create a validation for that during the sprint

File: app/helpers/comments_helper.rb (hunk 1 above)
Comment (jpaints3) [reply app/helpers/comments_helper.rb:1]: Ok sounds good, we'll probably figure out if it's helpful in testing

**Pull request merged**